features/nanostack/mbed-mesh-api/source/WisunInterface.cpp

```diff
@@ -627,13 +663,21 @@ mesh_error_t WisunInterface::stack_info_get(ws_stack_state_t *stack_info_ptr)
     stack_info_ptr->pan_id = stack_info.pan_id;
     stack_info_ptr->rsl_in = stack_info.rsl_in;
     stack_info_ptr->rsl_out = stack_info.rsl_out;
+    stack_info_ptr->device_min_sens = stack_info.device_min_sens;
     memcpy(stack_info_ptr->parent_addr, stack_info.parent, 16);
     memcpy(stack_info_ptr->global_addr, global_address, 16);
     memcpy(stack_info_ptr->link_local_addr, link_local_address, 16);
 
     return MESH_ERROR_NONE;
 }
 
+const ws_cca_threshold_table_t *WisunInterface::cca_threshold_table_get(void)
+{
+    const cca_threshold_table_s *cca_threshold_table = arm_nwk_get_cca_threshold_table(get_interface_id());
+
+    return (const ws_cca_threshold_table_t *)cca_threshold_table;
```

Comment:
Casting directly to different API structures creates binding that prevents changes to be made flexibly
perhaps have the ws_cca_threshold_table at class and return it with filled fields from the nano stack API

Reply:
Correct me if I am wrong.
Can we take pointer of **ws_cca_threshold_table** from NM and fill it back ?
Like this:
`mesh_error_t WisunInterface::cca_threshold_table_get(const ws_cca_threshold_table_t *table)
{
    const cca_threshold_table_s *cca_table = arm_nwk_get_cca_threshold_table(get_interface_id());

    table->number_of_channels = cca_table->number_of_channels;
    table->cca_threshold_table = cca_table->cca_threshold_table;
    
    return MESH_ERROR_NONE;
}`
And the NM should call like this:
`{
    ws_cca_threshold_table_t cca_table_info;

    if (ws_iface->cca_threshold_table_get(cca_table_info) == MESH_ERROR_NONE) {
        tr_info("Got CCA threshold table");
    }
}`